src/main/java/org/jboss/tools/intellij/stackanalysis/SaUtils.java

```diff
@@ -0,0 +1,150 @@
+/*******************************************************************************
+ * Copyright (c) 2021 Red Hat, Inc.
+ * Distributed under license by Red Hat, Inc. All rights reserved.
+ * This program is made available under the terms of the
+ * Eclipse Public License v2.0 which accompanies this distribution,
+ * and is available at http://www.eclipse.org/legal/epl-v20.html
+ *
+ * Contributors:
+ * Red Hat, Inc. - initial API and implementation
+ ******************************************************************************/
+package org.jboss.tools.intellij.stackanalysis;
+
+import com.google.gson.Gson;
+import com.google.gson.JsonObject;
+import com.intellij.openapi.fileEditor.FileEditorManager;
+import com.intellij.openapi.vfs.LocalFileSystem;
+import com.intellij.openapi.vfs.VirtualFile;
+import org.apache.commons.compress.archivers.tar.TarArchiveEntry;
+import org.apache.commons.compress.archivers.tar.TarArchiveInputStream;
+import org.apache.commons.compress.utils.IOUtils;
+import org.jboss.tools.intellij.analytics.Platform;
+import org.jboss.tools.intellij.analytics.PlatformDetectionException;
+
+import java.io.BufferedWriter;
+import java.io.FileOutputStream;
+import java.io.IOException;
+import java.io.OutputStreamWriter;
+import java.io.File;
+import java.io.Writer;
+import java.io.FileInputStream;
+import java.io.BufferedInputStream;
+import java.util.zip.GZIPInputStream;
+
+public class SaUtils {
+
+    /**
+     * <p>Get Stack Analysis Report for given manifest file using CLI.</p>
+     *
+     * @param filePath Path to target manifest file
+     *
+     * @return A JSONObject having SA Report.
+     *
+     * @throws IOException In case of process failure
+     * @throws InterruptedException In case of process failure
+     */
+    public JsonObject getReport(String filePath) throws IOException, InterruptedException {
+        // Get the SA report using CLI and return JSON data
+        return new Gson().fromJson(new SaProcessExecutor().performStackAnalysis(filePath), JsonObject.class);
+    }
+
+
+    /**
+     * <p>Open a custom editor window.</p>
+     *
+     * <p>The custom editor window will open a file which will have browser attached to it.</p>
+     *
+     * @param instance An instance of FileEditorManager.
+     * @param url URL to be opened in attached browser.
+     *
+     * @throws IOException In case of process failure
+     */
+    public void openCustomEditor(FileEditorManager instance, String url, String manifestName) throws IOException {
+
+        // Close custom editor if already opened in previous run
+        closeCustomEditor(instance);
```

Comment:
I think that needs to be removed because we should be able to display several reports since the change to dynamic temp file or we should close only the one that correspond to the same original file

Reply:
That can be done, I am following the VSCode trend, there we show new report on top of existing report. I can do this also. Yusuf WDYT?

Reply:
in vscode we always have only a single file open for the report. I think i would like to have the same behaviour here as well. multiple files will be confusing

Reply:
Yes but the way it is implemented today is that you can have a report for pom.xml and another to go.mod for example. So if I rerun the report for pom.xml I don't want to close the editor for go.mod

Reply:
if we can make sure that for pom.xml, only 1 tab appears and other tabs for other deps file then its ok. What i would not want is multiple tabs for pom.xml

Reply:
Okay I will change, for a particular manifest only one tab will be opened. just need one clarification, what if a project is having multiple manifests (ex <project>/requirements.txt and <project>/test/requirements.txt) or there are two project of same language in same workspace (ex <project 1>/requirements.txt and <project 2>/requirements.txt) in this case do we need to use same tab to show report or multiple?

Reply:
One tab per file so /a/pom.xml and /b/pom.xml should have different tabs

Reply:
It is done. PTAL
PFA screenshot for an example

![image](https://user-images.githubusercontent.com/63054462/133050584-0dfb0bd2-ec6e-435b-8afc-8ea3951cd6a8.png)